Fix Windows build and rerender for CUDA 11.2

carterbox opened the pull request:
Fixes the broken Windows builds by adding `--use-loca-env` to nvcc calls. https://github.com/conda-forge/nvcc-feedstock/issues/76 This library uses an old version of CMake CUDA find_library and a strange way of declaring the build artifacts, so the flag needs to be added via `-DCUDA_NVCC_FLAGS="--use-local-env"` instead of the [`CUDA_FLAGS`](https://cmake.org/cmake/help/latest/envvar/CUDAFLAGS.html) environment variable or by appending the option to the[ `CUDACXX`](https://cmake.org/cmake/help/latest/envvar/CUDACXX.html?highlight=cudacxx) environment variable.

Changes the build and install steps to call cmake instead of generator/build system directly. Switches generator to Ninja for all platforms. Based on this PR, Ninja builds seem significantly faster or approximately the same duration.

Adds a special case for 11.2 in the Windows build script because it was missing.

Adds checks for additional build artifacts.

Adds a strict `run_export`.

Changes the maintainers from @isuruf to myself and @conda-forge/pytorch-cpu.

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

Closes #15

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

carterbox commented:
```
'cl.exe' is not recognized as an internal or external command,
operable program or batch file.

```

carterbox commented:
Wow great. It's the `--use-loca-env` that is needed for nvcc. But this library uses an old CMake CUDA api, so the flag needs to be added via `-DCUDA_NVCC_FLAGS="--use-local-env"`

carterbox commented:
Waiting to see if I can get the PowerPC and ARM builds to not time out on one of the CI providers. I have tried Travis so far. Waiting for azure currently. Will try Circle next?

EDIT: May try to setup cross-compiling if emulated and native builds are too slow.

carterbox commented:
Abandoned powerpc and arm for this PR.

hmaarrfk commented:
I'm confused about the change of maintainers. Why remove Isuru?

carterbox commented:
@hmaarrfk, please see issue #15 which this PR closes. Isuru is a member of core anyways, so being listed in the maintainers section only adds more notifications for him, but not extra permission.

hmaarrfk commented:
Thanks is this ready according to you?

carterbox commented:
Yes. Please review.

hmaarrfk commented on .ci_support/migrations/cuda_112_ppc64le_aarch64.yaml:
```diff
@@ -0,0 +1,41 @@
+migrator_ts: 1645421738
```
Can we remove this file since the attempt didn't work

carterbox replied:
Good catch! I originally based this PR on #17, but since I didn't end up enabling those builds, I probably shouldn't have any artifacts from that PR.

hmaarrfk reviewed (approved):
Just a small nit. Are you able to merge yourself?

carterbox commented:
I don't have special permission outside staged-recipes.

carterbox committed:
DOC: Add note why powerpc and arm are not built

carterbox committed:
Fixes the broken Windows builds by adding --use-loca-env to nvcc calls.

conda-forge/nvcc-feedstock#76 This library uses an old version of CMake
CUDA find_library and a strange way of declaring the build artifacts, so
the flag needs to be added via -DCUDA_NVCC_FLAGS="--use-local-env" instead
of the CUDA_FLAGS environment variable or by appending the option to the
CUDACXX environment variable.

Changes the build and install steps to call cmake instead of
generator/build system directly. Switches generator to Ninja for all
platforms. Based on this PR, Ninja builds seem significantly faster or
approximately the same duration.

Adds a special case for 11.2 in the Windows build script because it was missing.

Adds checks for additional build artifacts.

Adds a strict run_export.

Changes the maintainers from @isuruf to myself and @conda-forge/pytorch-cpu.

carterbox committed:
MNT: Re-rendered with conda-build 3.21.9, conda-smithy 3.21.2, and conda-forge-pinning 2022.10.20.19.20.00

carterbox commented:
OK. I reapplied my changes to the recipe/ and conda-forge.yml on a fresh branch from the main branch then rerendered. There should be no residual changes from #17.

hmaarrfk commented:
Thanks
